vck-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/openid/PresentationFactory.kt

```diff
@@ -158,101 +147,78 @@ internal class PresentationFactory(
         jsonWebKeys: Collection<JsonWebKey>?,
         responseWillBeEncrypted: Boolean,
     ): CoseSigned<ByteArray> {
-        val sessionTranscript =
-            if (dcApiRequest != null) {
-                calcSessionTranscript(dcApiRequest, nonce, jsonWebKeys, responseWillBeEncrypted)
-            } else if (mdocGeneratedNonce != null && clientId != null && responseUrl != null) {
-                calcSessionTranscript(
-                    mdocGeneratedNonce,
-                    clientId,
-                    responseUrl,
-                    nonce
-                )
-            } else {
-                null
-            }
-
-        return if (sessionTranscript != null) {
-            run {
-                val deviceAuthentication = DeviceAuthentication(
-                    type = "DeviceAuthentication",
-                    sessionTranscript = sessionTranscript,
-                    docType = docType,
-                    namespaces = ByteStringWrapper(DeviceNameSpaces(mapOf()))
-                )
-                val deviceAuthenticationBytes = coseCompliantSerializer
-                    .encodeToByteArray(ByteStringWrapper(deviceAuthentication))
-                    .wrapInCborTag(24)
-                    .also {
-                        Napier.d("Device authentication signature input is ${it.encodeToString(Base16())}")
-                    }
-
-                signDeviceAuthDetached(
-                    protectedHeader = null,
-                    unprotectedHeader = null,
-                    payload = deviceAuthenticationBytes,
-                    serializer = ByteArraySerializer()
-                ).getOrElse {
-                    throw PresentationException(it)
-                }
-            }
+        val sessionTranscript: SessionTranscript = if (dcApiRequest != null) {
+            calcSessionTranscriptForDcApi(dcApiRequest.callingOrigin, nonce, jsonWebKeys, responseWillBeEncrypted)
```

Comment:
i really like the line-by-line, explicit parameter names from all of the above refactorings. this call would also benefit from that style